tests/Testcontainers.Tests/Unit/Containers/Unix/Modules/Databases/CosmosDbTestcontainerTest.cs

```diff
@@ -0,0 +1,42 @@
+namespace DotNet.Testcontainers.Tests.Unit
+{
+  using System.Net;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Tests.Fixtures;
+  using Microsoft.Azure.Cosmos;
+  using Xunit;
+
+  public static class CosmosDbTestcontainerTest
+  {
+    [Collection(nameof(Testcontainers))]
+    public sealed class ConnectionTests : IClassFixture<CosmosDbFixture>
+    {
+      private readonly CosmosDbFixture fixture;
+
+      public ConnectionTests(CosmosDbFixture fixture)
+      {
+        this.fixture = fixture;
+      }
+
+      private CosmosClientOptions Options => new CosmosClientOptions { HttpClientFactory = () => this.fixture.Container.HttpClient, ConnectionMode = ConnectionMode.Gateway, };
```

Comment:
In general, I would prefer to create the client in the fixture, but that is a bit cumbersome due to the HTTP client. What about this:

```csharp
[Collection(nameof(Testcontainers))]
public sealed class ConnectionTests : IClassFixture<CosmosDbFixture>, IDisposable
{
  private readonly HttpClient httpClient;

  private readonly CosmosClient cosmosClient;

  public ConnectionTests(CosmosDbFixture cosmosDbFixture)
    : this(cosmosDbFixture.Container.HttpClient, cosmosDbFixture.Container.ConnectionString)
  {
  }

  private ConnectionTests(HttpClient httpClient, string connectionString)
  {
    var cosmosClientOptions = new CosmosClientOptions();
    cosmosClientOptions.ConnectionMode = ConnectionMode.Gateway;
    cosmosClientOptions.HttpClientFactory = () => httpClient;
    this.httpClient = httpClient;
    this.cosmosClient = new CosmosClient(connectionString, cosmosClientOptions);
  }

  [Fact]
  public async Task ShouldEstablishConnection()
  {
    var accountProperties = await this.cosmosClient.ReadAccountAsync()
      .ConfigureAwait(false);

    Assert.Equal("localhost", accountProperties.Id);
  }

  [Fact]
  public async Task CreateDatabaseTest()
  {
    var databaseResponse = await this.cosmosClient.CreateDatabaseIfNotExistsAsync("db")
      .ConfigureAwait(false);

    Assert.Equal(HttpStatusCode.Created, databaseResponse.StatusCode);
  }

  public void Dispose()
  {
    this.cosmosClient.Dispose();
    this.httpClient.Dispose();
  }
}
```

Reply:
Yes, looks good.